### DISCUSSION TITLE
Is it possible to have a single config.nsdepcop at sln level

### DISCUSSION THREAD

**Discussion by kimpenhaus** [realvizu/NsDepCop#77] [Q&A] [answered]
Hi,

first of all thanks for your effort - I was just evaluating NsDepCop and it looks great.

Only thing I was wondering - to get it running I needed to put a config.nsdeopcop file (even if it is empty) on every project level directory. Is it possible to have it just in the root folder on sln level?

Assume you have following dir-tree - I currently needed to have it like

```
| - sln
| - config.nsdepcop
| - src
| --- apis
| ----- api1
| ------- config.nsdeopcop (no rules)
| ----- api2
| ------- config.nsdeopcop (no rules)
....
```

I would like to have it like (so I couldn't forget to put a config when adding projects)

```
| - sln
| - config.nsdepcop
| - src
| --- apis
| ----- api1
| ----- api2
....
```

Not sure if I did something wrong :) but it would be great if all rules could go to the config.nsdeopcop at root level.

Thanks,
Marcus

**Comment by realvizu** [comment, marked as answer]
Yes, that's correct. Currently, each project folder where you want to run the tool must include a config.nsdepcop file.

This requirement exists because the tool functions as a Roslyn analyzer, which operates within the [project scope](https://learn.microsoft.com/en-us/visualstudio/code-quality/roslyn-analyzers-overview?view=vs-2022#scope) and processes the config file as an [additional file](https://github.com/dotnet/roslyn/blob/main/docs/analyzers/Using%20Additional%20Files.md).

That said, it could be possible to extend NsDepCop to look for a solution-level config file if a project-level one is not present. I'll create an issue for this enhancement, and we'll see if someone is interested in implementing it.

**Comment by kimpenhaus** [comment]
@realvizu  thanks for your feedback 😄 

I guess I've found viable way to beahve it like I wanted without any changes.
 I am using a `Directory.Build.props` on top level having

```
    <ItemGroup>
        <AdditionalFiles Include="$(SolutionDir)config.nsdepcop" />
    </ItemGroup>
```    

in it.

That way I don't have to add it to each project (and maybe miss that)

Cheers,
m
